app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,67 @@
+TASKS = [
+  {id: 1, name: "Wash dishes", description: "Use soap and hot water"},
+  {id: 2, name: "Do laundry", description: "Wash and dry clothes"},
+  {id: 2, name: "Grocery shop", description: "Go to store and buy healthy food"}
+]
+
+
+class TasksController < ApplicationController
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+    id = params[:id].to_i
+    @task = Task.find_by(id: id)
+
+    if @task.nil?
+      render :notfound, status: :not_found
+    end
+  end
+
+  def complete
+    @task = Task.find_by(id: params[:id].to_i)
```

Comment:
What happens if the task isn't found? You should include error-handling code in all of these actions that look for a single task.